### PR TITLE
Feature/login signup with keycloak

### PR DESCRIPTION
**Pull request by Hairax**
The theme change for login and registration was not made

### CODE REVIEW

**Review by Miguelito2774 — APPROVED** [review]

File: apps/frontend/app/api/auth/[...nextauth]/route.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        session.roles = token.decoded.realm_access.roles;
+      }
+      session.error = token.error;
+      return session;
+    },
+  },
+};
+
+const handler = NextAuth(authOptions);
+export { handler as GET, handler as POST };
```
Comment (Miguelito2774) [inline apps/frontend/app/api/auth/[...nextauth]/route.ts]: add end-line

File: apps/frontend/app/api/auth/logout/route.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    try {
+      await fetch(url, { method: 'GET' });
+    } catch (err) {
+      console.error(err);
+      return new Response(null, { status: 500 });
+    }
+  }
+
+  return new Response(null, { status: 200 });
+}
```
Comment (Miguelito2774) [inline apps/frontend/app/api/auth/logout/route.ts]: add end-line

File: apps/frontend/app/utils/encryption.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  return encryptedString; 
+}
+
+export function decrypt(encryptedString) {
+    const secretKey =  'F6AM7R/6QDPm6UHYCMZL1ttPnmV2IHFP95wGF2OWKL4';
+    const cryptr = new Cryptr(secretKey);
+  
+    const text = cryptr.decrypt(encryptedString);
+    return text;
+  }
```
Comment (Miguelito2774) [inline apps/frontend/app/utils/encryption.js]: add end-line

File: apps/frontend/app/utils/sessionProviderWrapper.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+'use client';
+import React from 'react';
+
+import { SessionProvider } from 'next-auth/react';
+
+const SessionProviderWrapper = ({ children }) => {
+  return <SessionProvider>{children}</SessionProvider>;
+};
+
+export default SessionProviderWrapper;
```
Comment (Miguelito2774) [inline apps/frontend/app/utils/sessionProviderWrapper.js]: add end-line

File: apps/frontend/app/utils/sessionTokenAccessor.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+export async function getIdToken() {
+
+  const session = await getServerSession(authOptions);  
+  if(session){    
+    const idTokenDecrypted = decrypt(session.id_token)    
+    return idTokenDecrypted;
+  }
+  return null;
+}
```
Comment (Miguelito2774) [inline apps/frontend/app/utils/sessionTokenAccessor.ts]: add end-line

**Review by Miguelito2774 — CHANGES_REQUESTED** [review]
It is necessary to export the configuration and ensure that the realm is loaded correctly, as it acts as the configuration persistence in Keycloak. To achieve this, you should:

1. Configure a volume for the Keycloak service.
2. Adjust the startup command to include the loading of the configuration from a JSON file within the project.

**Review by Miguelito2774 — APPROVED** [review]

**Review by GastonGutierrezC — APPROVED** [review]

**Pull request merged**